tests/testthat/test-shinytest2-tm_outliers.R

```diff
@@ -0,0 +1,312 @@
+  })
+  teal.data::datanames(data) <- "CO2"
+  teal.data::join_keys(data) <- teal.data::join_keys(join_key("CO2", "CO2", "primary_key"))
+
+  vars <- teal.transform::choices_selected(
+    teal.transform::variable_choices(
+      data[["CO2"]],
+      c("Plant", "Type", "Treatment")
+    )
+  )
+
+  init_teal_app_driver(
+    data = data,
+    modules = tm_outliers(
+      outlier_var = list(
+        teal.transform::data_extract_spec(
+          dataname = "CO2",
+          select = teal.transform::select_spec(
+            label = "Select variable:",
+            choices = teal.transform::variable_choices(data[["CO2"]], c("conc", "uptake")),
+            selected = "uptake",
+            multiple = FALSE,
+            fixed = FALSE
+          )
+        )
+      ),
+      categorical_var = list(
+        teal.transform::data_extract_spec(
+          dataname = "CO2",
+          filter = teal.transform::filter_spec(
+            vars = vars,
+            choices = teal.transform::value_choices(data[["CO2"]], vars$selected),
+            selected = teal.transform::value_choices(data[["CO2"]], vars$selected),
+            multiple = TRUE
+          )
+        )
+      ),
+      ggplot2_args = list(
+        teal.widgets::ggplot2_args(
+          labs = list(subtitle = "Plot generated by Outliers Module")
+        )
+      )
+    )
+  )
+}
+
+testthat::test_that("e2e - tm_outliers: data parameter and module label is passed properly", {
+  skip_if_too_deep(5)
+
+  app_driver <- app_driver_tm_outlier()
+  app_driver$expect_no_shiny_error()
+
+  testthat::expect_equal(
+    app_driver$get_text("#teal-main_ui-root-active_tab > li.active > a"),
+    "Outliers Module"
+  )
+
+  encoding_dataset <- app_driver$get_text("#teal-main_ui-root-outliers_module .help-block")
+  testthat::expect_match(encoding_dataset, "Dataset:[\n ]*CO2", all = FALSE)
+  testthat::expect_match(encoding_dataset, "Outlier data points", all = FALSE)
+
+  app_driver$stop()
+})
+
+testthat::test_that("e2e - tm_outliers:
+  data extract spec elements are initialized with the default values
+  specified by outlier_var and categorical_var argument", {
+  skip_if_too_deep(5)
+
+  app_driver <- app_driver_tm_outlier()
+  app_driver$expect_no_shiny_error()
+
+  testthat::expect_identical(
+    app_driver$get_active_module_input("outlier_var-dataset_CO2_singleextract-select"),
+    "uptake"
+  )
+  app_driver$set_active_module_input("outlier_var-dataset_CO2_singleextract-select", "conc")
+  app_driver$expect_no_shiny_error()
+
+  testthat::expect_identical(
+    app_driver$get_active_module_input("categorical_var-dataset_CO2_singleextract-filter1-col"),
+    "Plant"
+  )
+  app_driver$set_active_module_input("categorical_var-dataset_CO2_singleextract-filter1-col", "Type")
+  app_driver$expect_no_shiny_error()
+
+  testthat::expect_identical(
+    app_driver$get_active_module_input("categorical_var-dataset_CO2_singleextract-filter1-vals"),
```

Comment:
this test is failing because of setting `Type` in filter by.
`app_driver$set_active_module_input("categorical_var-dataset_CO2_singleextract-filter1-col", "Type")`

I would suggest we should test default values separately or first and then set values.

Reply:
<img width="629" alt="image" src="https://github.com/insightsengineering/teal.modules.general/assets/6700955/a8acb895-f73d-482c-b553-9116ccaaa0a5">

Reply:
fixed - thanks ce4da519688047069959912a61625c36e2ee27a2